src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -325,13 +332,17 @@ private void WriteAllPending()
 
             if (Monitor.TryEnter(_contextLock))
             {
-                _writePending = false;
+                _postingWrite = false;
 
                 if (_nextWriteContext != null)
                 {
                     writingContext = _nextWriteContext;
                     _nextWriteContext = null;
                 }
+                else
+                {
+                    _ongoingWrites--;
```

Comment:
Since there's no explicit check as when incrementing, I'd like to have a `Debug.Assert` here checking that we're not going below 0.